dask_sql/physical/rel/logical/window.py

```diff
@@ -224,39 +210,32 @@ class DaskWindowPlugin(BaseRelPlugin):
         "row_number": None,  # That is the easiest one: we do not even need to have any windowing. We therefore threat it separately
         "$sum0": SumOperation(),
         "sum": SumOperation(),
-        # Is replaced by a sum and count by calcite: "avg": ExplodedOperation(AvgOperation()),
         "count": CountOperation(),
         "max": MaxOperation(),
         "min": MinOperation(),
         "single_value": FirstValueOperation(),
         "first_value": FirstValueOperation(),
         "last_value": LastValueOperation(),
+        "avg": AvgOperation(),
     }
 
     def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContainer:
         (dc,) = self.assert_inputs(rel, 1, context)
 
-        # During optimization, some constants might end up in an internal
-        # constant pool. We need to dereference them here, as they
-        # are treated as "normal" columns.
-        # Unfortunately they are only referenced by their index,
-        # (which come after the real columns), so we need
-        # to always substract the number of real columns.
-        constants = list(rel.getConstants())
-        constant_count_offset = len(dc.column_container.columns)
-
         # Output to the right field names right away
         field_names = rel.getRowType().getFieldNames()
-
-        for window in rel.getGroups():
+        input_column_count = len(dc.column_container.columns)
+        for window in rel.window().getGroups():
             dc = self._apply_window(
-                window, constants, constant_count_offset, dc, field_names, context
+                rel, window, input_column_count, dc, field_names, context
             )
 
         # Finally, fix the output schema if needed
         df = dc.df
         cc = dc.column_container
-
+        cc = cc.limit_to(
+            cc.columns[input_column_count:] + cc.columns[0:input_column_count]
+        )
```

Comment:
Maybe out of scope for this PR, but wondering if expanding `ColumnContainer.add` to allow us to insert columns at a specific index would allow us to remove column reorders like this (which IMO might be confusing to developers since the `limit_to` method name doesn't make the intent of this call obvious); if that were the case, we could add the columns to the `ColumnContainer` in the desired order in `_apply_window`